File: backend/app/models/exam.py
from sqlalchemy import Column, Integer, String, Boolean, Text, DateTime
from datetime import datetime, timezone
from ..database import Base


class Exam(Base):
    __tablename__ = "exams"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, nullable=False)
    eligibility = Column(String, nullable=False)
    exam_pattern = Column(String, nullable=False)
    syllabus = Column(Text, default="")
    preparation_strategy = Column(Text, default="")
    course_available = Column(Boolean, default=True)
    description = Column(Text, nullable=True)
    created_at = Column(DateTime, default=lambda: datetime.now(timezone.utc))

File: lib/models/exam.dart
class Exam {
  final int? id;
  final String name;
  final String eligibility;
  final String examPattern;
  final String syllabus;
  final String preparationStrategy;
  final bool courseAvailable;
  final String? description;

  Exam({
    this.id,
    required this.name,
    required this.eligibility,
    required this.examPattern,
    this.syllabus = '',
    this.preparationStrategy = '',
    this.courseAvailable = true,
    this.description,
  });

  factory Exam.fromJson(Map<String, dynamic> json) {
    return Exam(
      id: json['id'] as int?,
      name: json['name'] as String? ?? '',
      eligibility: json['eligibility'] as String? ?? '',
      examPattern: json['exam_pattern'] as String? ?? '',
      syllabus: json['syllabus'] as String? ?? '',
      preparationStrategy: json['preparation_strategy'] as String? ?? '',
      courseAvailable: json['course_available'] as bool? ?? true,
      description: json['description'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'eligibility': eligibility,
      'exam_pattern': examPattern,
      'syllabus': syllabus,
      'preparation_strategy': preparationStrategy,
      'course_available': courseAvailable,
      'description': description,
    };
  }
}
